### DISCUSSION TITLE
Nested layout lists

### DISCUSSION THREAD

**Discussion by shbchk** [ExpressionEngine/ExpressionEngine#4539] [Feature Requests]
### Suggest an idea for ExpressionEngine

Currently there is no way to use layout lists within layout lists, e.g. handle grids within channel entries (or go even deeper with relationships inside grids).

In [Stash](https://github.com/croxton/Stash/wiki/Lists#three-and-more-dimensional-lists) there were nested multidimensional lists with `context=""` parameter, but it seems Layouts have nothing like that. It would be extremely beneficial if ExpressionEngine's layout system could support nested layout lists.

**Comment by intoeetive** [comment]
Can you give an example of how this could work? (like a pseudo template code)

**Comment by shbchk** [comment]
Here we are setting a context for a list using `context="{entry_id}"`:

```
// news/index
{layout="news/_markup"}

{exp:channel:entries channel="news"}
	{layout:set:append name="titles"}{title}{/layout:set:append}
	{layout:set:append name="entry_ids"}{entry_id}{/layout:set:append}

	{grid}
		{layout:set:append name="grid_items" context="{entry_id}"}{grid:item}{/layout:set:append}
		{layout:set:append name="grid_prices" context="{entry_id}"}{grid:price}{/layout:set:append}
	{/grid}
{/exp:channel:entries}
```

And here we're using context and prefix to display it:
```
// news/_markup
{layout:titles}
	<h1>{value}</h1>
	{layout:grid_items context="{layout:entry_ids index='{index}'}" prefix="griditems"}
		<div>
			{griditems:value}
				/
			{layout:grid_prices context="{layout:entry_ids index='{index}'}" index="{griditems:index}"}
		</div> 
	{/layout} 
{/layout:titles}
```

Does it make any sense to you?
